### PR TITLE
feature: add the yamlint tool to check check all yaml files

### PR DESCRIPTION
**Pull request by fengzixu**
Sign-off-by: fengzixu  <hnustphoenix@gmail.com>

### Ⅰ. Describe what this PR did

I add the yamllint tool to check yaml files in Dragonfly, so that we can ensure that their format is legal.

### Ⅱ. Does this pull request fix one issue?

fix https://github.com/dragonflyoss/Dragonfly/issues/1112

### CODE REVIEW

**Comment by yeya24** [comment]
CI failed because of a bad link. I will file a PR for it.

**Comment by allencloud** [comment]
CI fails:
```
#!/bin/bash -eo pipefail
find  ./* -name  "*" | xargs misspell -error
xargs: misspell: No such file or directory

Exited with code exit status 127
```

I have tested the image `pouchcontainer/pouchlinter:v0.2.6`locally, there is no `misspell` in it:
```
~ # docker run -it pouchcontainer/pouchlinter:v0.2.6 bash
root@8c37e24d3126:/#
root@8c37e24d3126:/# which misspell
root@8c37e24d3126:/#
```

**Comment by fengzixu** [comment]
> CI fails:
> 
> ```
> #!/bin/bash -eo pipefail
> find  ./* -name  "*" | xargs misspell -error
> xargs: misspell: No such file or directory
> 
> Exited with code exit status 127
> ```
> 
> I have tested the image `pouchcontainer/pouchlinter:v0.2.6`locally, there is no `misspell` in it:
> 
> ```
> ~ # docker run -it pouchcontainer/pouchlinter:v0.2.6 bash
> root@8c37e24d3126:/#
> root@8c37e24d3126:/# which misspell
> root@8c37e24d3126:/#
> ```

I know. But when I build the image of pouchlinter's master branch on my local env, misspell works fine.

image
Is there any mistake during build pouchlinter:v0.2.6?

**Comment by fengzixu** [comment]
I run the build command in pouchlinter directory: `docker build . -t pouchlint`

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,6 +37,10 @@ jobs:
           name: use opensource tool client9/misspell to correct commonly misspelled English words
           command: |
             find  ./* -name  "*" | xargs misspell -error
+      - run:
+          name: use yamllint tool to check the format of all yaml files
+          command: |
+            find ./* -name "*" | xargs yamllint -f colored
```
Comment (allencloud) [inline .circleci/config.yml]: I think we just need to check the yaml file, so `find ./* -name "*.yml" | xargs yamllint -f colored` ?

**Comment by fengzixu** [comment]
image
Some illegal characters cause errors when yamllint tool performed

**Comment by fengzixu** [comment]
It looks like the `yamllint` works fine. But we need to check if some part of rules we need to ignore them.

**Comment by yeya24** [comment]
@fengzixu  Could you please fix all the warning and error?

**Comment by fengzixu** [comment]
> @fengzixu Could you please fix all the warning and error?

Doing

**Comment by fengzixu** [comment]
@yeya24 @allencloud 
I have formated all yaml files according to the suggestion of yamllint. 
BYW, I just ignore [line-length](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.line_length), because we cannot confirm the length 80 is suitable for our project.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**